.compatability_tests/compatibility_test_1_2/tests/integration/compatibility_1_2.rs

```diff
@@ -0,0 +1,94 @@
+#[test_case(test_roqoqo_1_2::operations::MeasureQubit::new(0,"ro".to_string(), 1).into(); "MeasureQubit")]
+#[test_case(test_roqoqo_1_2::operations::PragmaGetStateVector::new("ro".to_string(), None).into(); "PragmaGetStateVector")]
+#[test_case(test_roqoqo_1_2::operations::PragmaGetDensityMatrix::new("ro".to_string(), None).into(); "PragmaGetDensityMatrix")]
+#[test_case(test_roqoqo_1_2::operations::PragmaGetOccupationProbability::new("ro".to_string(), None).into(); "PragmaGetOccupationProbability")]
+#[test_case(test_roqoqo_1_2::operations::PragmaGetPauliProduct::new(std::collections::HashMap::new(),"ro".to_string(), test_roqoqo_1_2::Circuit::new()).into(); "PragmaGetPauliProduct")]
+#[test_case(test_roqoqo_1_2::operations::PragmaRepeatedMeasurement::new("ro".to_string(), 10, None).into(); "PragmaRepeatedMeasurement")]
+#[test_case(test_roqoqo_1_2::operations::PragmaDepolarising::new(0,1.0.into(), 1.0.into()).into(); "PragmaDepolarising")]
+#[test_case(test_roqoqo_1_2::operations::PragmaDamping::new(0,1.0.into(), 1.0.into()).into(); "PragmaDamping")]
+#[test_case(test_roqoqo_1_2::operations::PragmaDephasing::new(0,1.0.into(), 1.0.into()).into(); "PragmaDephasing")]
+#[test_case(test_roqoqo_1_2::operations::PragmaRandomNoise::new(0,1.0.into(), 1.0.into(), 1.0.into()).into(); "PragmaRandomNoise")]
+#[test_case(test_roqoqo_1_2::operations::PragmaActiveReset::new(0).into(); "PragmaActiveReset")]
+#[test_case(test_roqoqo_1_2::operations::PragmaConditional::new("ro".to_string(),0, test_roqoqo_1_2::Circuit::new()).into(); "PragmaConditional")]
+#[test_case(test_roqoqo_1_2::operations::MultiQubitMS::new(vec![0,2,3], 1.0.into()).into(); "MultiQubitMS")]
+#[test_case(test_roqoqo_1_2::operations::PhaseShiftedControlledPhase::new(0,1, 0.5.into(), 1.0.into()).into(); "PhaseShiftedControlledPhase")]
+#[test_case(test_roqoqo_1_2::operations::PragmaLoop::new(2.0.into(), test_roqoqo_1_2::Circuit::new()).into(); "PragmaLoop")]
+#[test_case(test_roqoqo_1_2::operations::InputBit::new("input".to_string(), 1, true).into(); "InputBit")]
+
+fn test_bincode_compatibility_1_2(operation: test_roqoqo_1_2::operations::Operation) {
+    let mut test_circuit = test_roqoqo_1_2::Circuit::new();
+    test_circuit += operation;
+
+    let test_measurement_input = test_roqoqo_1_2::measurements::PauliZProductInput::new(3, false);
+    let test_measurement = test_roqoqo_1_2::measurements::PauliZProduct {
+        constant_circuit: Some(test_circuit.clone()),
+        circuits: vec![test_circuit],
+        input: test_measurement_input,
+    };
+    let test_program = test_roqoqo_1_2::QuantumProgram::PauliZProduct {
+        measurement: test_measurement,
+        input_parameter_names: vec!["test".to_string()],
+    };
+    let test_serialisation: Vec<u8> = bincode::serialize(&test_program).unwrap();
+
+    let _test_deserialisation: roqoqo::QuantumProgram =
+        bincode::deserialize(&test_serialisation).unwrap();
```

Comment:
Should we check that the two QuantumPrograms are equal?

Reply:
We could extend the tests by this later, but it is unlikely that the deserialization works but by accident produces a wrong object.